examples/flutter_gallery/lib/demo/shrine/shrine_order.dart

```diff
@@ -0,0 +1,227 @@
+                          }).toList(),
+                          value: quantity,
+                          onChanged: quantityChanged
+                        )
+                      )
+                    )
+                  ),
+                  new SizedBox(height: 16.0),
+                  new SizedBox(
+                    height: 24.0,
+                    child: new Align(
+                      alignment: FractionalOffset.bottomLeft,
+                      child: new Text(product.vendor.name, style: theme.vendorTitleStyle)
+                    )
+                  ),
+                  new SizedBox(height: 16.0),
+                  new Text(product.vendor.description, style: theme.vendorStyle),
+                  new SizedBox(height: 24.0)
+                ]
+              )
+            )
+          ]
+        )
+      )
+    );
+  }
+}
+
+class OrderPage extends StatefulWidget {
+  OrderPage({ Key key, this.order, this.products }) : super(key: key) {
+    assert(order != null);
+    assert(products != null && products.length > 0);
+  }
+
+  final Order order;
+  final List<Product> products;
+
+  @override
+  _OrderPageState createState() => new _OrderPageState();
+}
+
+/// Displays a product's OrderItem above photos of all of the other products
+/// arranged in two columns. Enables used to specify a quantity and add an
+/// order to the shopping cart.
+class _OrderPageState extends State<OrderPage> {
+  final GlobalKey<ScaffoldState> scaffoldKey = new GlobalKey<ScaffoldState>(debugLabel: 'Order Page');
+
+  Order get currentOrder => ShrineOrderRoute.of(context).order;
+
+  set currentOrder(Order value) {
+    ShrineOrderRoute.of(context).order = value;
+  }
+
+  void updateOrder({ int quantity, bool inCart }) {
+    Order newOrder = currentOrder.copyWith(quantity: quantity, inCart: inCart);
+    if (currentOrder != newOrder) {
+      setState(() {
+        currentOrder = newOrder;
+      });
+    }
+  }
+
+  void showSnackBarMessage(String message) {
+    scaffoldKey.currentState.showSnackBar(new SnackBar(content: new Text(message)));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new ShrinePage(
+      scaffoldKey: scaffoldKey,
+      floatingActionButton: new FloatingActionButton(
+        onPressed: () {
+          updateOrder(inCart: true);
+          showSnackBarMessage('There are ${currentOrder.quantity} items in the shopping cart');
+        },
+        backgroundColor: const Color(0xFF16F0F0),
+        child: new Icon(
+          icon: Icons.add_shopping_cart,
+          color: Colors.black
+        )
+      ),
+      body: new Block(
+        children: <Widget>[
+          new OrderItem(
+            product: config.order.product,
+            quantity: currentOrder.quantity,
+            quantityChanged: (int value) { updateOrder(quantity: value); }
+          ),
+          new SizedBox(height: 24.0),
+          new FixedColumnCountGrid(
+            columnCount: 2,
```

Comment:
I have note tested on tablets, in fact I'm not supporting landscape layouts correctly yet.